src/gt4sd/algorithms/generation/diffusion/core.py

```diff
@@ -0,0 +1,353 @@
+            target=None,  # type:ignore
+        )
+
+    def get_generator(
+        self,
+        configuration: AlgorithmConfiguration[S, T],
+        target: Optional[T],
+    ) -> Untargeted:
+        """Get the function to sample batches.
+
+        Args:
+            configuration: helps to set up the application.
+            target: context or condition for the generation. Unused in the algorithm.
+
+        Returns:
+            callable generating a batch of items.
+        """
+        logger.info("ensure artifacts for the application are present.")
+        self.local_artifacts = configuration.ensure_artifacts()
+        implementation: Generator = configuration.get_conditional_generator(  # type: ignore
+            self.local_artifacts
+        )
+        return implementation.sample
+
+    def validate_configuration(
+        self, configuration: AlgorithmConfiguration
+    ) -> AlgorithmConfiguration:
+        assert isinstance(configuration, AlgorithmConfiguration)
+        return configuration
+
+
+@ApplicationsRegistry.register_algorithm_application(DiffusersGenerationAlgorithm)
+class DiffusersConfiguration(AlgorithmConfiguration[str, None]):
+    """Basic configuration for a diffusion algorithm."""
+
+    algorithm_type: ClassVar[str] = "generation"
+    domain: ClassVar[str] = "generic"
+
+    modality: str = field(
+        default="image",
+        metadata=dict(
+            description="Model domain.  Supported: 'image', 'text', 'audio', 'token2image', 'token2sample', 'molecule'."
+        ),
+    )
+
+    model_type: str = field(
+        default="diffusion",
+        metadata=dict(
+            description=f"Type of the model. Supported: {', '.join(MODEL_TYPES.keys())}"
+        ),
+    )
+
+    scheduler_type: str = field(
+        default="discrete",
+        metadata=dict(
+            description=f"Type of the noise scheduler. Supported: {', '.join(SCHEDULER_TYPES.keys())}"
+        ),
+    )
+
+    temperature: float = field(
+        default=1.0,
+        metadata=dict(
+            description="Temperature for sampling, the lower the greedier the sampling."
+        ),
+    )
+    repetition_penalty: float = field(
+        default=1.0,
+        metadata=dict(
+            description="Primarily useful for CTRL model, where 1.2 should be used."
+        ),
+    )
+    k: int = field(
+        default=50,
+        metadata=dict(description="Number of top-k probability tokens to keep."),
+    )
+    p: float = field(
+        default=1.0,
+        metadata=dict(
+            description="Only tokens with cumulative probabilities summing up to this value are kept."
+        ),
+    )
+    prefix: str = field(
+        default="",
+        metadata=dict(
+            description="Text defining context provided prior to the prompt."
+        ),
+    )
+    number_of_sequences: int = field(
+        default=8,
+        metadata=dict(description="Number of text sequences to generate."),
```

Comment:
does this refer only to text models or is this just `number_of_samples`?

Reply:
this is just for text. I remove it for the moment.